crates/runtime/src/vmcontext.rs

```diff
@@ -227,6 +227,14 @@ pub struct VMMemoryDefinition {
     pub current_length: AtomicUsize,
 }
 
+macro_rules! ensure {
+    ($cond:expr, $trap:expr) => {
+        if !($cond) {
+            return Err($trap);
+        }
+    };
+}
```

Comment:
Would it work to use [`anyhow::ensure`](https://docs.rs/anyhow/latest/anyhow/macro.ensure.html)?

Reply:
removed totally